Sofa/framework/Geometry/src/sofa/geometry/Edge.h

```diff
@@ -216,15 +216,15 @@ struct Edge
     * @tparam   T scalar
     * @param	pA, pB nodes of the first edge
     * @param	pC, pD nodes of the second edge
-    * @param    intersection node will be filled if there is an intersection otherwise will return std::numeric_limits<T>::min()
+    * @param    intersectionBaryCoord barycentric coordinates of the intersection point expressed as alpa * pA + beta * pB if there is an intersection ,  node will be filled if there is an intersection otherwise will return [0, 0]
     * @return	bool true if there is an intersection, otherwise false
     */
     template<typename Node,
         typename T = std::decay_t<decltype(*std::begin(std::declval<Node>()))>,
         typename = std::enable_if_t<std::is_scalar_v<T>>
     >
         [[nodiscard]]
-    static constexpr bool intersectionWithEdge(const Node& pA, const Node& pB, const Node& pC, const Node& pD, Node& intersection)
+    static constexpr bool intersectionWithEdge(const Node& pA, const Node& pB, const Node& pC, const Node& pD, sofa::type::Vec<2, T>& intersectionBaryCoord)
```

Comment:
Wouldn't it be more versatile to parametrize the `EQUALITY_THRESHOLD` ? I don't know if it is currently resulting from a error estimation or arbitrary set, but to be able to modify it would highly affect the result of this method. For instance we can set it as a default argument `eqThresh = EQUALITY_THRESHOLD` so anyone could change it depending on their need.

Reply:
I'm using the value define in Vec.h so it is shared by Mat / Vec operation. I don't know if this threshold should be easily changed?
Current value:
`#define EQUALITY_THRESHOLD 1e-6`